### PR TITLE
Bug 2174 2151 2445 release to develop

### PR DESCRIPTION
**Pull request by nitish-beehyv**
Bug #2174 #2151 #2445 release to develop

## Release Notes

- **New Features**
  - Improved error display with dynamic styling and a new back button component.
  - Enhanced localization for date-related messages in the review card.
  - Added new UI customization options for better handling of specific keys.
  - Updated delivery channel logic to ensure unique entries and conditional inclusion based on order type.
  - Refined submission handling and user role management for a smoother submission process.

- **Bug Fixes**
  - Corrected error handling and improved task assignment logic in submission processes.

### CODE REVIEW

**Review by vaibhavct — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorBoundaries.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ class ErrorBoundary extends React.Component {
       // ("UI-component-details", this.props);
       // You can render any custom fallback UI
       return (
-        <div className="error-boundary">
+        <div className="error-boundary" style={{ height: "calc(100vh - 84px)" }}>
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorBoundaries.js:41]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorBoundaries.js [hunk 2]
```diff
@@ -9,7 +9,7 @@
     process.env.NODE_ENV === "development"
   ) {
     //do nothing
-  }else{
+  } else {
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorBoundaries.js:12]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorComponent.js [hunk 3]
```diff
@@ -82,6 +99,8 @@
 };
 
 const ErrorComponent = (props) => {
+  const CustomButton = window?.Digit?.ComponentRegistryService?.getComponent("CustomButton");
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorComponent.js:102]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorComponent.js [hunk 4]
```diff
@@ -45,19 +44,37 @@
...
+      stroke-linejoin="round"
+    />
+    <path
+      d="M7.57031 1.09025C5.45497 3.20559 4.26899 4.39158 2.15365 6.50692L7.57031 11.9236"
+      stroke="#3A3A3A"
+      stroke-width="2.5769"
+      stroke-linejoin="round"
+    />
+  </svg>
+);
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorComponent.js:76]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/orders/src/components/SummonsOrderComponent.js [hunk 5]
```diff
@@ -273,8 +273,8 @@ const SummonsOrderComponent = ({ t, config, formData, onSelect, clearErrors }) =
       const ePostAddresses = addressList?.filter((item) => Boolean(item));
       setDeliveryChannels(
         [
-          { label: "SMS", type: "SMS", code: "SMS", values: phone_numbers || [] },
-          { label: "EMAIL", type: "E-mail", code: "EMAIL", values: email || [] },
+          { label: "SMS", type: "SMS", code: "SMS", values: [...new Set(phone_numbers || [])] },
+          { label: "EMAIL", type: "E-mail", code: "EMAIL", values: [...new Set(email || [])] },
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/orders/src/components/SummonsOrderComponent.js:277]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/configs/UICustomizations.js [hunk 6]
```diff
@@ -312,6 +312,8 @@ export const UICustomizations = {
       switch (key) {
         case "CASE_TYPE":
           return <span>NIA S138</span>;
+        case "CS_STAGE":
+          return t(value);
         case "CS_SCRUTINY_STATUS":
           return t(row?.status === "UNDER_SCRUTINY" ? "IN_PROGRESS" : "NOT_STARTED");
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/configs/UICustomizations.js:318]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionsCreate.js [hunk 7]
```diff
@@ -756,12 +756,15 @@ const SubmissionsCreate = ({ path }) => {
               action: "CREATE",
             },
             filingNumber: response?.application?.filingNumber,
-            assignedTo: response?.application?.additionalDetails?.respondingParty?.flatMap((item) => item?.uuid?.map((u) => ({ uuid: u }))),
+            assignedTo: response?.application?.additionalDetails?.respondingParty
+              ?.flatMap((item) => item?.uuid?.map((u) => ({ uuid: u })))
+              ?.filter((item) => item?.uuid !== userInfo?.uuid),
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionsCreate.js:761]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorComponent.js [hunk 8]
```diff
@@ -120,14 +139,15 @@ const ErrorComponent = (props) => {
...
             props.goToHome();
           }}
+          icon={<LeftBackArrow />}
+          label={t(config.buttonInfo)}
           style={styles.button}
-        >
-          {t(config.buttonInfo)}
-        </button>
+          textStyles={styles.buttonTextStyle}
+        />
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/components/ErrorComponent.js:150]: checked

**Pull request merged**